tests/ui/cast_size.stderr

```diff
@@ -6,25 +6,27 @@ LL |     1isize as i8;
    |
    = note: `-D clippy::cast-possible-truncation` implied by `-D warnings`
 
-error: casting isize to f64 causes a loss of precision on targets with 64-bit wide pointers (isize is 64 bits wide, but f64's mantissa is only 52 bits wide)
+error: casting integer literal to f64 is unnecessary
   --> $DIR/cast_size.rs:12:5
    |
 LL |     1isize as f64;
-   |     ^^^^^^^^^^^^^
+   |     ^^^^^^^^^^^^^ help: try: `1_f64`
    |
-   = note: `-D clippy::cast-precision-loss` implied by `-D warnings`
+   = note: `-D clippy::unnecessary-cast` implied by `-D warnings`
 
-error: casting usize to f64 causes a loss of precision on targets with 64-bit wide pointers (usize is 64 bits wide, but f64's mantissa is only 52 bits wide)
+error: casting integer literal to f64 is unnecessary
```

Comment:
This warning shouldn't change, as for example 9999999999999999_usize != 9999999999999999_f64

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f72692b7a11575384abd2b6fbc95599f)

Reply:
Yes, it is.
But 9999999999999999_usize as f64 == 9999999999999999_f64, so the suggestion equal to orgin code.

Maybe we should check mantissa bits wide for integer literal case.
Then add the test case mentioned above.
Is it ok?

Reply:
Maybe we should rewrite some test case.
The error type doesn't match name of test case.

Reply:
I fixed some test case to keep original error type.

Reply:
> 9999999999999999_usize as f64 == 9999999999999999_f64

That is correct, but the previous warning explicitly warns you about this kind of cast, since you lose precision with it. The new suggestion would indicate to the programmer, that this cast is always ok and he can change the code.